### PR TITLE
RDSK-8056

### PR DESCRIPTION
**Pull request by arush-mehrotra**
**_Jira_**: [https://viam.atlassian.net/browse/RSDK-8056](https://viam.atlassian.net/browse/RSDK-8056)

**_Changes Made_**: The purpose of this change is to ensure that no files of size 0 bytes are uploaded. Instead, an error is thrown. Binary and tabular data are uploaded via `upload_data_capture_file.go` in which there is a pre-existing check ensuring that only files with `len(sensorData) > 0` are uploaded and synced. However, this check did not exist within `upload_arbitrary_file.go`, allowing users to upload empty files. Thus, this change covers that edge case. 

**_Testing_**:  I have conducted manual testing by attempting to sync a file of 0 bytes and ensuring the proper error messages are logged and that the file is not synced or deleted from the local disk. I have also added a unit test to ensure an empty file returns an error.

### CODE REVIEW

File: services/datamanager/builtin/sync_test.go [hunk 1]
```diff
@@ -332,6 +332,7 @@ func TestArbitraryFileUpload(t *testing.T) {
 	datasync.RetryExponentialFactor.Store(int32(1))
 	fileName := "some_file_name.txt"
 	fileExt := ".txt"
+	emptyFileTestName := "error due to empty file, local files should not be deleted"
```
Comment (kim-mishra) [inline services/datamanager/builtin/sync_test.go:335]: why not define this in line like all of the other test names?

File: services/datamanager/builtin/sync_test.go [hunk 2]
```diff
@@ -403,7 +410,9 @@ func TestArbitraryFileUpload(t *testing.T) {
 			// Write file to the path.
 			var fileContents []byte
 			for i := 0; i < 1000; i++ {
-				fileContents = append(fileContents, []byte("happy cows come from california\n")...)
+				if tc.name != emptyFileTestName {
+					fileContents = append(fileContents, []byte("happy cows come from california\n")...)
+				}
```
Comment (kim-mishra) [inline services/datamanager/builtin/sync_test.go:415]: might be nice to pull out this for loop into a helper called `populateFileContents` or something and only call it `if tc.name != emptyFileTestName` so you don't loop 1000 times for nothing

**Review by kim-mishra — APPROVED** [review]
had some nits but otherwise LGTM, thanks for this!

File: services/datamanager/builtin/sync_test.go (hunk 1 above)
Comment (arush-mehrotra) [reply services/datamanager/builtin/sync_test.go:335]: main reason was just because when I do the special casing in the actual test logic, i didn't want the check to be hardcoded (`if tc.name == "error due to empty file, local files should not be deleted"`) in case the test name needed to be changed for any reason. 

if you think it's better to hard code it though, i'm definitely open to changing it!

File: services/datamanager/builtin/sync_test.go (hunk 2 above)
Comment (arush-mehrotra) [reply services/datamanager/builtin/sync_test.go]: yeah I agree - just fixed it!

File: services/datamanager/builtin/sync_test.go (hunk 1 above)
Comment (kim-mishra) [reply services/datamanager/builtin/sync_test.go:335]: i think thats a good reason!

**Review by vijayvuyyuru — APPROVED** [review]
LGTM, thanks for implementing.
